async_simple/Executor.h

```diff
@@ -91,12 +91,20 @@ class Executor {
     // `schedule` would return false if schedule failed, which means function
     // func will not be executed. In case schedule return true, the executor
     // should guarantee that the func would be executed.
-    virtual bool schedule(Func func) = 0;
+    //
+    // The priority_hint is used to guide the scheduling order, larger number
+    // indicating higher priority. Default priority for most async-simple class
+    // is 0. The scheduling logic isn't necessary, which is determined by
+    // implementation. However, to avoid spinlock deadlock, when priority_hint
+    // is less than 0, scheduler shouldn't always execute the work immediately.
+    virtual bool schedule(Func func,
+                          [[maybe_unused]] int priority_hint = 0) = 0;
```

Comment:
Do not force users to implement this, leave a default implementation to `schedule.`